trading_calendars/exchange_calendar_asex.py

```diff
@@ -0,0 +1,168 @@
+
+AssumptionDay = assumption_day()
+
+NationalHoliday2 = Holiday(
+    'National Holiday 2',
+    month=10,
+    day=28,
+)
+
+ChristmasEve = christmas_eve(start_date='2009')
+
+ChristmasDay = christmas()
+
+SecondDayOfChristmas = Holiday(
+    'Second Day of Christmas (w/ no added Friday off)',
+    month=12,
+    day=26,
+)
+
+
+class ASEXExchangeCalendar(TradingCalendar):
+    """
+    Exchange calendar for the Athens Stock Exchange (ASEX).
+
+    Open Time: 10:00 AM, EEST
+    Close Time: 5:25 PM, EEST
+
+    Regularly-Observed Holidays:
+    - New Year's Day
+    - Epiphany (Jan 6)
+    - Orthodox Ash Monday (48 days before Orthodox Easter Sunday)
+    - National Holiday (Mar 25)
+    - Good Friday
+    - Easter Monday
+    - Orthodox Good Friday
+    - Orthodox Easter Monday
+    - Labour Day (May 1)
+    - Orthodox Whit Monday (50 days after Orthodox Easter Sunday)
+    - Assumption Day (Aug 15)
+    - National Holiday (Oct 28)
+    - Christmas Eve
+    - Christmas Day
+    - Second Day of Christmas (Dec 26)
+
+    Early Closes:
+    - None
+    """
+    name = 'ASEX'
+
+    tz = timezone('Europe/Athens')
+
+    open_times = (
+        (None, time(10, 1)),
+    )
+
+    close_times = (
+        (None, time(17, 25)),
+    )
+
+    @property
+    def regular_holidays(self):
+        return HolidayCalendar([
+            NewYearsDay,
+            Epiphany,
+            NationalHoliday1,
+            GoodFriday,
+            EasterMonday,
+            LabourDay,
+            AssumptionDay,
+            NationalHoliday2,
+            ChristmasEve,
+            ChristmasDay,
+            SecondDayOfChristmas
+        ])
+
+    @property
+    def adhoc_holidays(self):
+        debt_crisis = pd.date_range('2015-06-29', '2015-07-31', freq='B')
+
+        debt_crisis_holidays = [
+            pd.Timestamp(str(date), tz=UTC) for date in debt_crisis
+        ]
+
+        misc_adhoc_holidays = [
+            # In 2002, market closed for unknown reason
+            pd.Timestamp('2002-05-07', tz=UTC),
+            # In 2004, Assumption Day fell on a sunday, observed on a Friday
+            pd.Timestamp('2004-08-13', tz=UTC),
+            # In 2008, worker strikes closed the market for 2 days in March
+            pd.Timestamp('2008-03-04', tz=UTC),
+            pd.Timestamp('2008-03-05', tz=UTC),
+            # In 2013, May Day strikes closed the market
+            pd.Timestamp('2013-05-07', tz=UTC),
+            # In 2014, New Year's Eve was observed as a holiday
+            pd.Timestamp('2014-12-31', tz=UTC),
+            # In 2016, Labour Day fell on a Sunday, observed on Tuesday
+            pd.Timestamp('2016-05-03', tz=UTC),
+        ]
+
+        all_manually_computed = [
+            debt_crisis_holidays,
+            misc_adhoc_holidays,
+            list(OrthodoxGoodFriday),
+            list(OrthodoxEasterMonday),
+            list(OrthodoxWhitMonday),
+            list(OrthodoxAshMonday),
```

Comment:
These don't really feel like adhoc seeing as they're regularly observed by a set rule. I don't know if this would be better, but is there an easy way to define `OrthodoxEaster` as a `Holiday` instead of a datetime index, and then define the other Orthodox holidays as an offset to `OrthodoxEaster`? (similar to how we do `offset=[Easter(), Day(n)]` in places)

Reply:
Agreed, but I haven't been able to figure out a good way around this.  The info for Orthodox Easter is given in `dateutil.easter`, and using it to create a `Holiday` would require a new `Offset` class (https://github.com/pandas-dev/pandas/blob/master/pandas/tseries/offsets.py) which looks pretty involved (just looking at the one they use for regular Easter).

Reply:
Ok, in that case given that it could be pretty involved maybe we could just leave a comment here saying like `TODO: Investigate making this an actual Holiday`.